examples/contract_human_readable.rs

```diff
@@ -19,13 +17,30 @@ abigen!(
 
 #[tokio::main]
 async fn main() -> Result<()> {
-    // 1. compile the contract (note this requires that you are inside the `examples` directory) and
-    // launch ganache
-    let (compiled, ganache) =
-        compile_and_launch_ganache(Solc::new("**/contract.sol"), Ganache::new()).await?;
-    let contract = compiled.get("SimpleStorage").expect("could not find contract");
+    // the directory we use is root-dir/examples
+    let root = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("examples");
+    // we use `root` for both the project root and for where to search for contracts since
+    // everything is in the same directory
+    let paths = ProjectPathsConfig::builder().root(&root).sources(&root).build().unwrap();
+    // get the solc project instance using the paths above
+    let solc = Project::builder()
+        .paths(paths)
+        .ephemeral()
+        .artifacts(ArtifactOutput::Nothing)
+        .build()
+        .unwrap();
+    // compile the project and get the artifacts
+    let compiled = solc.compile().unwrap();
+    let compiled = match compiled {
+        ProjectCompileOutput::Compiled((output, _)) => output,
+        _ => panic!("expected compilation artifacts"),
+    };
```

Comment:
I feel like we should think about either:

* adding an error type that treats a CompilerOutput with a hard error as `Err(CompilerOutput)`, this will be more convenient for build scripts and gets rid of the `println!("{}", output);` to print diagnostics
* and an `CompilerOutput::into_result` function